### PR TITLE
Add health endpoints

### PR DESCRIPTION
**Pull request by josefkarasek**
## Add health probes

New http server is started with two handlers: ready and live. Both endpoints right now always return `200 ok`.

## Related issue(s)

https://github.com/kubeshop/botkube/issues/881
https://cloud.google.com/kubernetes-engine/docs/concepts/ingress#interpreted_hc

### CODE REVIEW

File: cmd/botkube/main.go [hunk 1]
```diff
@@ -386,6 +394,19 @@ func newMetricsServer(log logrus.FieldLogger, metricsPort string) *httpsrv.Serve
...
 }
 
+type healthChecker struct{}
+
+func (healthChecker) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
+	fmt.Fprint(resp, "ok")
+}
+
+func newHealthProbesServer(log logrus.FieldLogger, port string) *httpsrv.Server {
+	mux := mux.NewRouter()
```
[the remark this answers is not in the document]
Comment (huseyinbabal) [reply cmd/botkube/main.go:404]: I don't think creating a separate mux router is a good idea, what about having as follows

```
func newMetricsServer(log logrus.FieldLogger, metricsPort string) *httpsrv.Server {
	addr := fmt.Sprintf(":%s", metricsPort)
	router := mux.NewRouter()
	router.Handle("/metrics", promhttp.Handler())
        router.Handle(readinessEndpointName, healthChecker{})
        router.Handle(livenessEndpointName, healthChecker{})
	return httpsrv.New(log, addr, router)
}
```

**Comment by huseyinbabal** [comment]
@josefkarasek to make this pr work, we need to add some default values to our helm chart. Otherwise, even we have health endpoint, it will not be functional

File: pkg/config/config_test.go [hunk 2]
```diff
@@ -26,6 +26,7 @@ func TestLoadConfigSuccess(t *testing.T) {
 	t.Setenv("BOTKUBE_SETTINGS_CLUSTER__NAME", "cluster-name-from-env")
 	t.Setenv("BOTKUBE_SETTINGS_KUBECONFIG", "kubeconfig-from-env")
 	t.Setenv("BOTKUBE_SETTINGS_METRICS__PORT", "1313")
+	t.Setenv("BOTKUBE_SETTINGS_HEALTH__PORT", "1414")
```
Comment (huseyinbabal) [inline pkg/config/config_test.go:29]: Once we add health endpoint under same router, we won't need dedicated port for that

File: pkg/config/testdata/TestLoadConfigSuccess/config.golden.yaml [hunk 3]
```diff
@@ -444,6 +444,7 @@ settings:
             configMap:
                 name: runtime-config
     metricsPort: "1313"
+    healthPort: "1414"
```
Comment (huseyinbabal) [inline pkg/config/testdata/TestLoadConfigSuccess/config.golden.yaml:447]: same as above

File: pkg/execute/config_test.go [hunk 4]
```diff
@@ -60,6 +60,7 @@ func TestConfigExecutorShowConfig(t *testing.T) {
 						            fileName: ""
 						            configMap: {}
 						    metricsPort: ""
+						    healthPort: ""
```
Comment (huseyinbabal) [inline pkg/execute/config_test.go:63]: same as above

File: pkg/config/config_test.go (hunk 2 above)
Comment (josefkarasek) [reply pkg/config/config_test.go:29]: Any suggestion how to do the change?

1. Keep `BOTKUBE_SETTINGS_METRICS__PORT` and document it also covers health port configuration
2. Deprecate `BOTKUBE_SETTINGS_METRICS__PORT` and introduce more generic envar, eg `BOTKUBE_SETTINGS_OBSERV__PORT`

Comment (pkosiec) [reply pkg/config/config_test.go]: 🤔 That would be a breaking change, not sure if this is needed. Honestly I would stick to a separate server for now 👍 Yes, later we can have a single port for all the endpoints (GraphQL, metrics, health ones).

**Review by huseyinbabal — APPROVED** [review]

**Pull request merged**